File: bench/cross/vdot.ts
// Dot product of two 2M-element vectors — idiomatic TypeScript.
const n = 2000000;
const a = new Float64Array(n);
for (let i = 0; i < n; i++) a[i] = i;
let total = 0;
for (let i = 0; i < n; i++) total += a[i] * a[i];
console.log((total / 1e15).toFixed(3));

File: bench/cross/vdot.rs
// Dot product of two 2M-element vectors — idiomatic single-thread Rust.
fn main() {
    let n = 2_000_000usize;
    let a: Vec<f64> = (0..n).map(|i| i as f64).collect();
    let total: f64 = a.iter().map(|x| x * x).sum();
    println!("{:.3}", total / 1e15);
}
